solr/core/src/java/org/apache/solr/handler/admin/api/SnapshotBackupAPI.java

```diff
@@ -0,0 +1,174 @@
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.handler.ReplicationHandler;
+import org.apache.solr.handler.ReplicationHandler.ReplicationHandlerConfig;
+import org.apache.solr.jersey.JacksonReflectMapWriter;
+import org.apache.solr.jersey.PermissionName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** V2 endpoint for Backup API used for User-Managed clusters and Single-Node Installation. */
+@Path("/cores/{coreName}/replication/backups")
+public class SnapshotBackupAPI extends JerseyResource {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final SolrCore solrCore;
+  private final ReplicationHandlerConfig replicationHandlerConfig;
+
+  @Inject
+  public SnapshotBackupAPI(SolrCore solrCore, ReplicationHandlerConfig replicationHandlerConfig) {
+    this.solrCore = solrCore;
+    this.replicationHandlerConfig = replicationHandlerConfig;
+  }
+
+  /**
+   * This API (POST /api/cores/coreName/replication/backups {...}) is analogous to the v1
+   * /solr/coreName/replication?command=backup
+   */
+  @POST
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @Operation(summary = "Backup command using ReplicationHandler")
+  @PermissionName(CORE_EDIT_PERM)
+  public BackupReplicationResponse createBackup(
+      @RequestBody BackupReplicationRequestBody backupReplicationPayload) throws Exception {
+    if (backupReplicationPayload == null) {
+      throw new SolrException(
+          SolrException.ErrorCode.BAD_REQUEST, "Required request-body is missing");
+    }
+    ReplicationHandler replicationHandler =
+        (ReplicationHandler) solrCore.getRequestHandler(ReplicationHandler.PATH);
+    return doBackup(replicationHandler, backupReplicationPayload);
+  }
+
+  private BackupReplicationResponse doBackup(
+      ReplicationHandler replicationHandler,
+      BackupReplicationRequestBody backupReplicationPayload) {
+    BackupReplicationResponse response = instantiateJerseyResponse(BackupReplicationResponse.class);
+    int numberToKeep = backupReplicationPayload.numberToKeep;
+    int numberBackupsToKeep = replicationHandlerConfig.getNumberBackupsToKeep();
+    String location = backupReplicationPayload.location;
+    String repoName = backupReplicationPayload.repository;
+    String commitName = backupReplicationPayload.commitName;
+    String name = backupReplicationPayload.name;
+    Consumer<NamedList<?>> resultConsumer = result -> response.result = result;
+    try {
+      ReplicationHandler.doSnapShoot(
```

Comment:
[0] It'd prob be easier to test this API in `SnapshotBackupAPITest` if the underling functionality lived in some helper class (that I guess was injected into the ctor).  Not sure it's worth the hassle and noise of refactoring that all, but figured I'd mention it here in case you liked the idea 🤷

Reply:
updated the test, introduced a test class with a dummy impl so it simplifies the verification a bit.